cmd/sovereignnode/incomingHeader/interface.go

```diff
@@ -4,7 +4,7 @@ import "github.com/multiversx/mx-chain-core-go/data"
 
 // HeadersPool should be able to add new headers in pool
 type HeadersPool interface {
-	AddHeader(headerHash []byte, header data.HeaderHandler)
+	AddHeaderInShard(headerHash []byte, header data.HeaderHandler, shardID uint32)
```

Comment:
I think this is chainID

Reply:
No, should be shardID.
We fixed (I hope) every shard ID inconsistency. Everything coming from mainchain will be stored/saved  with a`MainChainShardID`